### PR TITLE
Add OMIT_MODULE_EXPORTS to omit module code

### PR DESCRIPTION
**Pull request by kjlubick**
This export code is the root cause of https://github.com/flutter/flutter/issues/58428 so we would like to be able to remove it.

### CODE REVIEW

**Comment by sbc100** [comment]
I would love to find a way to do this without adding a new settings if we can.    However, I'm not familiar enough with MODULARIZE or the its users.     Maybe there are other options such as completely removing the UMD style export stuff?  

Does the `EXPORT_ES6` mode also cause you problems?

If we are going to add a new option I think we need a better name because both `MODULE` and `EXPORT` are pretty overloaded here.    Maybe something more specific like `NO_UMD_EXPORTS`?

@RReverser has been battling with modularize recently so might have some insight into the right direction to go here.

**Comment by kjlubick** [comment]
ES6 modules do work for the browser, but they make use on node.js more problematic (unless we want multiple builds, one for node, one for web).

I'm fine with renaming the option to NO_UMD_EXPORTS so as not to be confusing with the Module object.

File: emcc.py [hunk 1]
```diff
@@ -2874,19 +2874,19 @@ def modularize():
...
-    elif not shared.Settings.MINIMAL_RUNTIME:
-      f.write('''\
-if (typeof exports === 'object' && typeof module === 'object')
-  module.exports = %(EXPORT_NAME)s;
-else if (typeof define === 'function' && define['amd'])
-  define([], function() { return %(EXPORT_NAME)s; });
-else if (typeof exports === 'object')
-  exports["%(EXPORT_NAME)s"] = %(EXPORT_NAME)s;
-''' % {'EXPORT_NAME': shared.Settings.EXPORT_NAME})
+    if not shard.Settings.OMIT_MODULE_EXPORTS:
```
Comment (kripken) [inline emcc.py:2877]:
```suggestion
    if not shared.Settings.OMIT_MODULE_EXPORTS:
```

**Review by kripken — COMMENTED** [review]
@kjlubick For your use case, can you build with `EXPORT_ES6` (which would avoid emitting UMD code)?

**Comment by kripken** [comment]
Oh sorry, I just saw the last comment now. So ES6 modules don't work in node. Is there no way to ignore them there, or is it a syntax error?

**Comment by kjlubick** [comment]
AFAIK, there is no way to ignore the export command in node.js

**Comment by RReverser** [comment]
Looking at https://github.com/flutter/flutter/issues/58428, it seems that your issue is rather that it's an anonymous `define`?

I thought these should work fine with Require.js, but, if not, maybe a better fix would be for us to generate some name there?

**Comment by RReverser** [comment]
> So ES6 modules don't work in node. Is there no way to ignore them there, or is it a syntax error?

Wait, ES6 modules do work in Node nowadays, and few versions ago got even unflagged. Can you try with latest Node.js version and set Emscripten's output extension to `.mjs` (which implies `EXPORT_ES6`)?

**Pull request closed without merging**